### PR TITLE
Support after_deploy() migrations after the delay has passed

### LINKED ISSUE #51: Support automating Safe.after_deploy migrations
I'd like to solve the problem of running Safe.after_deploy afterwards where I’m waiting to confirm a problem didn’t occur first. In these cases, I tend to forget to migrate those commands so I’d like to automate it.

One idea is to have a new model that tracks when a migration is detected. Then create a new management command that would need to be called periodically that would use a `timedelta` on the `Safe.after_deploy` migrations to determine which migrations can be migrated in the background.

From Ryan:

> 
> I can imagine this including some configuration in the migration itself to determine what kind of mandatory delay is required before running it. That kinda sounds like it could be a workable idea. What if we
> 1. make a `post_migrate` signal receiver to notice when it's first seen and marked as "ready" (not blocked) by safemigrate  .
> 2. have the `pre_migrate` signal compare the required delay and the (possible) record in the database to see if it's ready to go. If so, it would automatically convert to an always.
> 
> Maybe the new marking could something like
> ```safe = timedelta(days=7)```
> 
> That would avoid a new property. It would be an alternative to `Safe.after_deploy`. It doesn't generalize nicely, since there's no good meaning in that data type that would map to "before_deploy only" that I can see just yet.

Continuing on that thread, maybe we go with `safe = Safe.after_deploy(delay=timedelta(days=7))` to maintain consistency and avoid a typecheck.

### PR DESCRIPTION
**Pull request by tim-schilling**
This introduces backwards incompatible change in the enum is replaced with class methods: `Safe.before_deploy` -> `Safe.before_deploy()`

A migration with a `Safe.after_deploy(delay=timedelta(days=1))` will be considered as a `Safe.before_deploy` when a day has passed since it was first detected. To automatically run `Safe.after_deploy()` migrations, a command can be setup to run `safemigrate` periodically. 

Closes #51

### CODE REVIEW

File: src/django_safemigrate/__init__.py [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
+        return cls(safe=SafeEnum.before_deploy)
+
+    @classmethod
+    def after_deploy(cls, *, delay: timedelta = None):
+        return cls(safe=SafeEnum.after_deploy, delay=delay)
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return other.safe == self.safe and other.delay == self.delay
+        raise TypeError()
```
Comment (ryanhiebert) [inline src/django_safemigrate/__init__.py:33]: Is there something inappropriate about the automatically generated `__eq__` method?

**Review by ryanhiebert — COMMENTED** [review]
This is great. Do you think it would be reasonable to add some backward compatibility for the most expected usage by special-casing handling the `before_deploy`, `after_deploy`, and `always` methods, and automatically call them without arguments if they are detected?

That way `safe = Safe.before_deploy` would still work. It would still be backward-incompatible, but I think it would be a significantly easier transition.

File: src/django_safemigrate/__init__.py (hunk 1 above)
Comment (tim-schilling) [reply src/django_safemigrate/__init__.py:33]: Nope, I can remove it since we're not using it.

Comment (ryanhiebert) [reply src/django_safemigrate/__init__.py:33]: Sounds good. My impressions was that the autogenerated one should work pretty well even if we were.

**Comment by tim-schilling** [comment]
> Do you think it would be reasonable to add some backward compatibility for the most expected usage by special-casing handling the before_deploy, after_deploy, and always methods, and automatically call them without arguments if they are detected?

I think in an ideal world, yes. Is that something you have the time/interest to add in?

> That way safe = Safe.before_deploy would still work. It would still be backward-incompatible, but I think it would be a significantly easier transition.

The transition is find and replace `safe = Safe.before_deploy` with `safe = Safe.before_deploy()`, then do the same for `after_deploy` and `always`. It's not super burdensome, though it would be nice to remove it.

**Comment by ryanhiebert** [comment]
> I think in an ideal world, yes. Is that something you have the time/interest to add in?

Yeah, I'll see about getting to it in the next couple of days. I'm debating whether to make it just a backward compatibility thing, or follow a pretty typical Django convention to allow it to be either a value or a callable, like the `default` argument to model fields. I'm leaning toward the latter right now.

**Comment by ryanhiebert** [comment]
If this PR has a clean test suite, we can either merge this anticipating that follow-up, or I can update this PR with that improvement before we merge.

**Comment by tim-schilling** [comment]
@ryanhiebert alright, I've got everything passing now. May I bother you for a green checkmark?

**Review by ryanhiebert — APPROVED** [review]

**Pull request merged**